### PR TITLE
Add isAuthorized

### PR DESCRIPTION
**Pull request by cedricbazoge**
include unit test

### CODE REVIEW

File: PrivacyKit/Shared/PrivacyManager.swift [hunk 1]
```diff
@@ -39,5 +39,25 @@ public final class PrivacyManager {
         return .authorized
     }
     
+    // Return true if all purposeID in the list are authorized
+    // If any is unknown, we default to false.
+    public func isAuthorized(for purposeIDs:[String]) -> Bool {
+        
+        guard let consent = dataSource?.userConsents else {
+            return false
```
Comment (nidalfakhouri) [inline PrivacyKit/Shared/PrivacyManager.swift]: can we call this 'userConsents' to keep things syntactically aligned.

File: PrivacyKit/Shared/PrivacyManager.swift [hunk 2]
```diff
@@ -39,5 +39,25 @@ public final class PrivacyManager {
         return .authorized
     }
     
+    // Return true if all purposeID in the list are authorized
+    // If any is unknown, we default to false.
+    public func isAuthorized(for purposeIDs:[String]) -> Bool {
```
Comment (nidalfakhouri) [inline PrivacyKit/Shared/PrivacyManager.swift]: why would we want to pass in multiple purpose ids?

File: PrivacyKit/Shared/PrivacyManager.swift [hunk 3]
```diff
@@ -39,5 +39,25 @@ public final class PrivacyManager {
...
+        
+        guard let consent = dataSource?.userConsents else {
+            return false
+        }
+        // If any the purpose identifier passed in parameter is not
+        // included in our list of user consent we return false by default
+        for identifier in purposeIDs{
+            guard consent.contains(where: { $0.purposeIdentifier == identifier } ) else {
```
Comment (nidalfakhouri) [inline PrivacyKit/Shared/PrivacyManager.swift]: can you put a space between purposeIDs and the {

File: PrivacyKit/Shared/PrivacyManager.swift [hunk 4]
```diff
@@ -39,5 +39,25 @@ public final class PrivacyManager {
...
+                return false
+            }
+        }
+        // If consent does not have permission for at least one of
+        // the purpose. We are not authorized
+        return !consent
+            .contains( where: { $0.permission != .authorized } )
+    }
+
```
Comment (nidalfakhouri) [inline PrivacyKit/Shared/PrivacyManager.swift]: I am not sure I understand the logic here, the way I am reading it is that the purpose identifier does not have to align with the id passed into this method to return true. I feel like we need to be matching the identifiers passed in to the identifiers in the user consent object for this test to be valid.

File: PrivacyKit/Shared/PersonalData.swift [hunk 5]
```diff
@@ -26,13 +26,15 @@ class PersonalData <Element> {
...
+    init (_ item:Element?, purpose identifiers:[String]){
         self._item = item
-        self.purposeIds = IDs
+        self.purposeIdentifiers = identifiers
     }
     
+    // Determine if item is available according to the list of
+    // purpose identifier attached to it.
     private func purposeAvailable() -> Bool {
-        //loop thru purposeIds and check against the purpose manager if they are authorized
```
Comment (nidalfakhouri) [inline PrivacyKit/Shared/PersonalData.swift]: personally I don't think available is the correct terminology here. I think enabled is the more proper verbiage.

File: PrivacyKit/Shared/PrivacyManager.swift (hunk 2 above)
Comment (cedricbazoge) [reply PrivacyKit/Shared/PrivacyManager.swift]: that more convenient than having to call multiple times if a data covert several purpose

**Pull request merged**